pal/src/misc/random.cpp

```diff
@@ -17,35 +17,54 @@ Module Name:
 
 --*/
 
-#include "pal/palinternal.h"
-#include <sodium.h>
+#ifdef _CRT_RAND_S
 
-/*++
-Function :
-RANDOMInitialize
+#include "pal/palinternal.h"
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <mutex>
 
-Initialize libsodium
 
-(no parameters)
+#define RANDOM_CACHE_SIZE 8
```

Comment:
`constexpr` lets compiler run some basic logic during compile time while `define` is just an alias